File: examples/comparison-example.ts
import { ChilePayments } from '../src';

// Ejemplo mostrando cómo usar ambos proveedores con el mismo código
async function comparisonExample() {
  console.log('=== API Unificada - Mismo código, diferentes proveedores ===\n');

  const providers: Array<'webpay' | 'getnet'> = ['webpay', 'getnet'];

  for (const provider of providers) {
    console.log(`\n--- Probando ${provider.toUpperCase()} ---`);

    const payments = new ChilePayments({
      provider,
      environment: 'test',
    });

    try {
      const transaction = await payments.createTransaction({
        amount: 5000,
        orderId: `ORDEN-${provider}-${Date.now()}`,
        returnUrl: 'http://localhost:3000/return',
        description: `Pago de prueba con ${provider}`,
      });

      console.log('✅ Transacción creada exitosamente');
      console.log('Proveedor:', payments.getProvider());
      console.log('Token:', transaction.token);
      console.log('URL de pago:', transaction.paymentUrl);
      console.log('Estado:', transaction.status);

    } catch (error: any) {
      console.error('❌ Error:', error.message);
    }
  }

  console.log('\n✨ ¡El mismo código funciona con ambos proveedores!');
}

comparisonExample();

File: kotlin/examples/src/main/kotlin/cl/payments/examples/ComparisonExample.kt
package cl.payments.examples.comparison

import cl.payments.ChilePayments
import cl.payments.types.Environment
import cl.payments.types.PaymentProvider
import cl.payments.types.SDKConfig
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    println("=== API Unificada - Mismo código, diferentes proveedores ===\n")

    val providers = listOf(PaymentProvider.WEBPAY, PaymentProvider.GETNET)

    for (provider in providers) {
        println("\n--- Probando ${provider.name} ---")

        val payments = ChilePayments(
            SDKConfig(
                provider = provider,
                environment = Environment.TEST
            )
        )

        try {
            val transaction = payments.createTransaction(
                amount = 5000,
                orderId = "ORDEN-${provider.name}-${System.currentTimeMillis()}",
                returnUrl = "http://localhost:3000/return",
                description = "Pago de prueba con ${provider.name}"
            )

            println("✅ Transacción creada exitosamente")
            println("Proveedor: ${payments.getProvider()}")
            println("Token: ${transaction.token}")
            println("URL de pago: ${transaction.paymentUrl}")
            println("Estado: ${transaction.status}")

        } catch (e: Exception) {
            println("❌ Error: ${e.message}")
        }
    }

    println("\n✨ ¡El mismo código funciona con ambos proveedores!")
}
